### PR TITLE
auto generate api docs and fix accordingly

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR updates the api docs and makes small adjustments to the code base where necessary.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/aea.py [hunk 1]
```diff
@@ -34,12 +35,15 @@
 from aea.protocols.default.message import DefaultMessage
 from aea.registries.filter import Filter
 from aea.registries.resources import Resources
-from aea.skills.error import ERROR_SKILL_ID
 from aea.skills.error.handlers import ErrorHandler
 from aea.skills.tasks import TaskManager
 
 logger = logging.getLogger(__name__)
 
+ERROR_SKILL_ID = PublicId(
```
Comment (DavidMinarsch) [inline aea/aea.py:43]: There was a circular import issue with this. Now looking at it more carefully, I think we need to make this configurable anyway. The user should be forced to specify the error handler (by default we use ours).

File: aea/connections/base.py [hunk 2]
```diff
@@ -34,9 +34,6 @@
     from aea.mail.base import Envelope, Address  # pragma: no cover
 
 
-logger = logging.getLogger(__name__)
```
Comment (DavidMinarsch) [inline aea/connections/base.py:37]: unused

File: aea/connections/scaffold/connection.py [hunk 3]
```diff
@@ -19,14 +19,11 @@
 
 """Scaffold connection and channel."""
 
-import logging
 from typing import Optional
 
 from aea.connections.base import Connection
 from aea.mail.base import Envelope
 
-logger = logging.getLogger(__name__)
```
Comment (DavidMinarsch) [inline aea/connections/scaffold/connection.py:28]: unused

File: aea/skills/error/__init__.py [hunk 4]
```diff
@@ -18,6 +18,3 @@
 # ------------------------------------------------------------------------------
 
 """This module contains the implementation of the error skill."""
-from aea.configurations.base import PublicId
```
Comment (DavidMinarsch) [inline aea/skills/error/__init__.py:21]: this caused issues

Comment (DavidMinarsch) [reply aea/skills/error/__init__.py:21]: It is also not particularly neat, as the constant easily can get out of line with the config file.

Comment (DavidMinarsch) [reply aea/skills/error/__init__.py:21]: Ideally, the public id would be constructed from the config file.

File: aea/skills/error/handlers.py [hunk 5]
```diff
@@ -29,8 +29,6 @@
 from aea.protocols.default.serialization import DefaultSerializer
 from aea.skills.base import Handler
 
-logger = logging.getLogger(__name__)
```
Comment (DavidMinarsch) [inline aea/skills/error/handlers.py:32]: moved to context based logger

File: aea/aea.py (hunk 1 above)
Comment (marcofavorito) [reply aea/aea.py:43]: Ok, but maybe not. The error skill should be the same for every agent (although we can't enforce that anyway).

File: aea/skills/error/__init__.py (hunk 4 above)
Comment (marcofavorito) [reply aea/skills/error/__init__.py:21]: yes, it is how it works now. The public id is retrieved from the configuration object.

**Pull request merged**